### PR TITLE
Generic dependencies: refactor remote dependencies

### LINKED ISSUE #755: development generic dependencies
update so the user specifies an image (allowing for using name-spaced images) and a name.

```yml
# current 
dependencies: 
  # (mongo3.0.4 becomes the service name) 
  - name: mongo
    version: 3.0.4

# new
dependencies: 
  # (my-custom will be the service name)
  my-custom:
    image: 'mongo:0.26.4'
```

Also note that switching to using an object means we will need to sort by the dependency name in order to ensure out docker-compose file stays consistent.

@alexdavid please review to see if this captures the issue you ran into yesterday

### PR DESCRIPTION
**Pull request by hugoesthere**
Partially resolves: https://github.com/Originate/exosphere/issues/755

### CODE REVIEW

File: src/application/deployer/index.go [hunk 1]
```diff
@@ -92,23 +92,19 @@ func ValidateConfigs(deployConfig deploy.Config) error {
...
 		if err != nil {
 			return err
 		}
-		validatedDependencies[dependency.Name] = dependency.Version
 	}
 
 	fmt.Fprintln(deployConfig.Writer, "Validating service dependencies...")
 	for _, serviceContext := range deployConfig.AppContext.ServiceContexts {
 		for _, dependency := range serviceContext.Config.Remote.Dependencies {
-			if validatedDependencies[dependency.Name] == "" {
```
Comment (hugoesthere) [inline src/application/deployer/index.go:107]: I don't think this check is needed anymore

File: src/config/remote_generic_dependency.go [hunk 2]
```diff
@@ -12,24 +10,18 @@ type remoteGenericDependency struct {
...
 // GetDockerConfig returns docker configuration and an error if any
 func (g *remoteGenericDependency) GetDockerConfig() (types.DockerConfig, error) {
-	return types.DockerConfig{
-		Image: fmt.Sprintf("%s:%s", g.config.Name, g.config.Version),
-	}, nil
-}
-
-func (g *remoteGenericDependency) GetServiceName() string {
-	return g.config.Name + g.config.Version
+	return types.DockerConfig{}, nil
```
Comment (hugoesthere) [inline src/config/remote_generic_dependency.go:18]: We aren't using generic remote dependencies. When we do support them, we have to define a `docker-image` field somewhere under `remote.dependencies`, and specify that remote generic dependencies are for docker type dependencies only

Comment (charlierudolph) [reply src/config/remote_generic_dependency.go:18]: Lets just get rid of remote generic dependencies then and if the type isn't exocom/rds we throw an error

**Review by charlierudolph — APPROVED** [review]

File: src/config/remote_generic_dependency.go (hunk 2 above)
Comment (hugoesthere) [reply src/config/remote_generic_dependency.go:18]: Hmm, we'd be breaking a lot of apis if we started returning errors with `NewRemoteAppDependency`. Do you think it makes sense to keep remote generic dependency as a placeholder until we define what it does exactly?

Comment (charlierudolph) [reply src/config/remote_generic_dependency.go:18]: Okay. Good to leave as is for now since this will be changing soon

**Pull request merged**